### DISCUSSION TITLE
how do I customize shared dict memory

### DISCUSSION THREAD

**Discussion by qclucky7** [Kong/kong#9127] [Help] [answered]
I want to customize a plugin, but there are some caches that I don't want to use kong.cache, I want to use custom, is there a good way, and how can I use it in the plugin。thank you。

**Comment by chronolaw** [comment, marked as answer]
You can inject Nginx directive into nginx.conf, reference is here: https://docs.konghq.com/gateway/latest/reference/configuration/#injecting-nginx-directives

**Comment by qclucky7** [comment]
Thank you for your answer, I get it. I looked at the use of multi-tier caching in Kong, I defined a custom shared dictionary memory, how I use it in plugins, similar to the Kong.cache :get() API provided by Kong. or do I need to maintain the use of the cache myself? thank you。

**Reply by chronolaw** [reply]
I think you should use ngx_lua's API `ngx.shared.DICT`, https://github.com/openresty/lua-nginx-module#ngxshareddict.

**Reply by qclucky7** [reply]
I get it. thanks

**Reply by Tieske** [reply]
@GravityMatrix you might want to have a look at https://github.com/thibaultcha/lua-resty-mlcache for more complex caching

**Reply by qclucky7** [reply]
> @GravityMatrix you might want to have a look at https://github.com/thibaultcha/lua-resty-mlcache for more complex caching
Thank you,  I'll look at it, because I need to store the table structure, shared dict not support,  need to serialize and deserialize it every time, so local caching is better.
